### PR TITLE
[Solidity] Add line number decoding for counterexample trace

### PR DESCRIPTION
**Pull request by kunjsong01**
1. Added line number support. 
2. Dynamic array fixes and merge.

ESBMC-Solidity is now able to give exact line number in the couterexample.

### CODE REVIEW

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@kunjsong01: Could you please add information about the line number in the test.desc file?

**Comment by kunjsong01** [comment]
>Could you please add information about the line number in the test.desc file?

All test.desc files have been updated to include the flag `--contract <solidity source file>` for line number indications. 

Do we also want to add some additional information in test.desc for line number indication? What kind of information would it be?

File: src/langapi/language_ui.cpp [hunk 1]
```diff
@@ -73,7 +73,20 @@ bool language_uit::parse(const std::string &filename)
...
     language.set_func_name(_cmdline.vm["function"].as<std::string>());
+
+    if(config.options.get_option("contract") == "")
+    {
+      msg.error("Please set the smart contract source file.");
+      return true;
+    }
+    else
+    {
+      //language.set_smart_contract_source(_cmdline.vm["contract"].as<std::string>());
```
Comment (lucasccordeiro) [inline src/langapi/language_ui.cpp]: Remove this.

File: regression/esbmc-solidity/github_dynamic_loop/test.desc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CORE
+dynamic_array_oob_loop.solast
+--function dyn_array_oob_loop --incremental-bmc --contract dynamic_array_oob_loop.sol
+^VERIFICATION FAILED$
```
Comment (lucasccordeiro) [inline regression/esbmc-solidity/github_dynamic_loop/test.desc:5]: Add the line number in the regular expression.

File: src/solidity-frontend/pattern_check.cpp [hunk 3]
```diff
@@ -44,6 +45,7 @@ bool pattern_checker::start_pattern_based_check(const nlohmann::json &func)
 void pattern_checker::check_authorization_through_tx_origin(
   const nlohmann::json &func)
 {
+  printf("DEBUG_TX_ORIGIN: in check_authorization_through_tx_origin\n");
```
Comment (lucasccordeiro) [inline src/solidity-frontend/pattern_check.cpp]: Do we want to keep this debug function?

File: src/solidity-frontend/solidity_convert.cpp [hunk 4]
```diff
@@ -315,6 +334,10 @@ bool solidity_convertert::get_function_definition(
   std::string name, id;
   get_function_definition_name(ast_node, name, id);
 
+  // TODO: debug remove before commit
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.cpp]: Remember removing this.

File: src/solidity-frontend/solidity_convert.h [hunk 5]
```diff
@@ -11,26 +11,33 @@
 #include <solidity-frontend/solidity_grammar.h>
 #include <solidity-frontend/pattern_check.h>
 
+#include <iostream> // @@ DEBUG: remove before commit
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.h]: Remove this?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@kunjsong01: once you address these comments, I'm happy to accept this PR.

**Comment by kunjsong01** [comment]
Added line numbers in all applicable test cases. 
Also removed all debug prints.

**Pull request merged**